### ISSUE TITLE
No HTTPS instance

### ISSUE THREAD

**Issue by alesribic73** [Ylianst/MeshCentral#3300]
How to start MeshCentral running without SSL? I only need plain HTTP on port 80?

**Comment by Ylianst** [comment]
If you are just looking to remove security completely, that is not supported as it's not secure and MeshCentral deals with a lot of sensitive data.

If you are looking to setup MeshCentral with a reverse proxy, take a look at section 16 on the [User's Guide](https://meshcentral.com/info/docs/MeshCentral2UserGuide.pdf) or check out my [video on setting up MeshCentral with NGINX](https://www.youtube.com/watch?v=YSmiLyKSX2I).

Basically, you need to add this line in the settings section of the config.json:

```
"TlsOffload": "127.0.0.1"
```

This will remove TLS from port 443 and make it non-TLS, but it will also assume that TLS is being done by another entity. The "127.0.0.1" is the IP address of the reverse proxy so that the real address of the endpoint is trusted when "X-Forward-For" header is received.

Hope that helps,
Ylian

**Comment by alesribic73** [comment]
I managed to finally enable Proxy with secure connection and settings you provided here. But any of agents connects.

**Comment by Ylianst** [comment]
If none of the agents connect, you can add this line in the `settings` section of the config.json:

```
"ignoreagenthashcheck": true
```

This should make the agent connect, but it is BAD FOR SECURITY. The better way is to add the following line in the domain section of the config.json:

```
"certurl": "https://127.0.0.1:443"
```

You need to put a HTTPS URL that will allow MeshCentral to load the TLS certificate from NGINX. So, the URL you provide should load the same TLS certificate that the agents will see.

Let me know if that works,
Ylian

**Comment by alesribic73** [comment]
This did not work for me. But i solve the issue with generating certicicates with Meshcentral.
